docs: Improved dotnet github action (#29769)
dotnet/docs: docs/devops/create-dotnet-github-action.md

For more information, see [Metadata syntax for GitHub Actions](https://docs.github.com/actions/creating-actions/metadata-syntax-for-github-actions).

### Pre-defined environment variables

> With Github Actions you'll get a lot of [Environment variables](https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables) by default. For instance the variable `GITHUB_REF` will always contain a reference to the branch or tag that triggered the workflow run. `GITHUB_REPOSITORY` has the owner and repository name, eg. `dotnet/docs`.

Reviewer: ```suggestion
With GitHub Actions, you'll get a lot of [environment variables](https://docs.github.com/actions/learn-github-actions/environment-variables#default-environment-variables) by default. For instance, the variable `GITHUB_REF` will always contain a reference to the branch or tag that triggered the workflow run. `GITHUB_REPOSITORY` has the owner and repository name, for example, `dotnet/docs`.
```

> With GitBub Actions, you'll get a lot of [environment variables](https://docs.github.com/actions/learn-github-actions/environment-variables#default-environment-variables) by default. For instance, the variable `GITHUB_REF` will always contain a reference to the branch or tag that triggered the workflow run. `GITHUB_REPOSITORY` has the owner and repository name, for example, `dotnet/docs`.

Reviewer 2: ```suggestion
With GitHub Actions, you'll get a lot of [environment variables](https://docs.github.com/actions/learn-github-actions/environment-variables#default-environment-variables) by default. For instance, the variable `GITHUB_REF` will always contain a reference to the branch or tag that triggered the workflow run. `GITHUB_REPOSITORY` has the owner and repository name, for example, `dotnet/docs`.
```

> You should explore the already available variables and use them accordingly.

Reviewer 2: ```suggestion
You should explore the pre-defined environment variables and use them accordingly.
```

:::image type="content" source="media/action-log.png" lightbox="media/action-log.png" border="true" alt-text=".NET code metrics - GitHub Action log":::

## Performance improvements

> If you followed along the sample you might have noticed that every time this workflow is run, it will do a **docker build** for that image. So every user is faced with some time to build the container before running it. Before releasing your github action to the marketplace, you should:

Reviewer: ```suggestion
If you followed along the sample, you might have noticed that every time this workflow is run, it will do a **docker build** for that image. So, every user is faced with some time to build the container before running it. Before releasing your GitHub action to the marketplace, you should:
```

> If you followed along the sample, you might have noticed that every time this action is used, it will do a **docker build** for that image. So, every user is faced with some time to build the container before running it. Before releasing your GitHub Action to the marketplace, you should:

Reviewer 2: ```suggestion
If you followed along the sample, you might have noticed that every time this action is used, it will do a **docker build** for that image. So, every trigger is faced with some time to build the container before running it. Before releasing your GitHub Action to the marketplace, you should:
```

1. (automatically) Build the Docker image

> 2. Push the docker image to Github container registry (or any other public container registry for that matter)

Reviewer: ```suggestion
2. Push the docker image to GitHub container registry (or any other public container registry for that matter)
```

> 2. Push the docker image to GitHub Container Registry (or any other public container registry for that matter)

Reviewer 2: ```suggestion
2. Push the docker image to the GitHub Container Registry (or any other public container registry)
```

3. Change the action to not build the image, but to use an image from a public registry.

```yaml
# Rest of action.yml content removed for readability
# using Dockerfile
runs:
  using: 'docker'
  image: 'Dockerfile' # Change this line
# using container image from public registry
runs:
  using: 'docker'
  image: 'docker://ghcr.io/some-user/some-registry' # Starting with docker:// is important!!
```

> 

Reviewer 2: ```suggestion

For more information, see [GitHub Docs: Working with the Container registry](https://docs.github.com/packages/working-with-a-github-packages-registry/working-with-the-container-registry).
```

## See also

- [.NET Generic Host](../core/extensions/generic-host.md)
- [Dependency injection in .NET](../core/extensions/dependency-injection.md)
- [DevOps for ASP.NET Core Developers](../architecture/devops-for-aspnet-developers/index.md)
- [Code metrics values](/visualstudio/code-quality/code-metrics-values)

> - [Open-source Github Action build in .NET](https://github.com/svrooij/dotnet-feeder/) with a [workflow](https://github.com/svrooij/dotnet-feeder/blob/main/.github/workflows/build.yml) for building and pushing the docker image automatically.

Reviewer 2: ```suggestion
- [Open-source GitHub Action build in .NET](https://github.com/svrooij/dotnet-feeder/) with a [workflow](https://github.com/svrooij/dotnet-feeder/blob/main/.github/workflows/build.yml) for building and pushing the docker image automatically.
```
